pkg/prometheus/promcfg.go

```diff
@@ -341,6 +355,13 @@ func (cg *configGenerator) generatePodMonitorConfig(version semver.Version,
 		},
 	}
 
+	// If timestamps honoring features are not set, then don't enforce anything.
+	// This is here to preserve backwards compatibility with prometheus <2.9.0
+	if ep.HonorTimestamps != nil || overrideHonorTimestamps {
```

Comment:
I would just pass all the values to honorTimestamps and handle everything in that function.

Reply:
This is mainly so the unit test you added covers these cases as well.

Reply:
I didn't have any idea on how to easily do that as this function would need to return a tristate value. Two ways I could think of how this could be done the way you suggest is to either:
1) return two values
2) initialize a helper variable before executing a function and then passing a pointer to that variable as a function parameter.

The way it is written right now, `honorTimestamps` handles all cases for prometheus >=2.9.0 and this check is only to ensure compatibility with prometheus <2.9.0. Unit tests handle all 6 outcome values of `honorTimestamps`.

Both resulted in a more complicated codebase. If you have a better idea of how this can be handled, then I am open to suggestions.

Reply:
Why can’t you pass in the cfg?